src/vfs/ftpfs/ftpfs.c

```diff
@@ -374,6 +374,9 @@ ftpfs_correct_url_parameters (const vfs_path_element_t *velement)
         // If user is different, remove password
         if (new_user != NULL && strcmp (path_element->user, new_user) != 0)
             MC_PTR_FREE (path_element->password);
+        else
+            // otherwise, use it
+            path_element->password = g_strdup (new_passwd);
```

Comment:
So in the end, the patch becomes like that, agree?

```suggestion
        // Use the password if it matches the user
        if (new_user != NULL && strcmp (path_element->user, new_user) == 0)
            path_element->password = g_strdup (new_passwd);
```

Reply:
I confirm this version passes the three test cases I described in https://github.com/MidnightCommander/mc/pull/4831#discussion_r2470776424

Re the other implementation details (string duplication etc), I'll trust you, because my C knowledge is quite limited.